java/vector/src/main/java/org/apache/arrow/vector/BitVectorHelper.java

```diff
@@ -140,22 +140,23 @@ public static int getNullCount(final ArrowBuf validityBuffer, final int valueCou
     }
     int count = 0;
     final int sizeInBytes = getValidityBufferSize(valueCount);
+    final int remainder = valueCount % 8;
 
-    for (int i = 0; i < sizeInBytes; ++i) {
-      final byte byteValue = validityBuffer.getByte(i);
-      /* Java uses two's complement binary representation, hence 11111111_b which is -1
-       * when converted to Int will have 32bits set to 1. Masking the MSB and then
-       * adding it back solves the issue.
-       */
-      count += Integer.bitCount(byteValue & 0x7F) - (byteValue >> 7);
+    for (int i = 0; i < sizeInBytes - 1; i++) {
```

Comment:
it's minor, but to be slightly more optimized maybe avoid the `- 1` in the loop condition and do something like:
```
final int sizeInBytesMinus1 = sizeInBytes - 1;
for (int i = 0; i < sizeInBytesMinus1; ++i) {
    ...
```